src/hio.c

```diff
@@ -400,7 +400,28 @@ HIO_HANDLE *hio_open_mem(const void *ptr, long size)
 		return NULL;
 
 	h->type = HIO_HANDLE_TYPE_MEMORY;
-	h->handle.mem = mopen(ptr, size);
+	h->handle.mem = mopen(ptr, size, 0);
+	h->size = size;
+
+	if (!h->handle.mem) {
+		free(h);
+		h = NULL;
+	}
+
+	return h;
+}
+
+HIO_HANDLE *hio_open_mem2(const void *ptr, long size)
```

Comment:
It looks like `hio_open_mem` is used in a grand total of 2 places in load.c currently... might be able to get away with just making this take a `free_after_use` param like `mopen`.